### ISSUE TITLE
Take inspiration from `goreleaser`

### ISSUE THREAD

**Issue by MarcoIeni** [MarcoIeni/release-plz#327]
## Motivations

[goreleaser](https://goreleaser.com/) is a tool for the go ecosystem that does a lot of automation around releases, such as building binaries, docker images, packages (aur, snap, brew, etc.), and announcements.

I haven't tried it but looks awesome! So kudos to the authors.

## Solution

I want to analyse all the features it has and create GitHub issues for the ones we want to implement in release-plz.

**Comment by brainstorm** [comment]
And perhaps [`cargo-dist`](https://www.reddit.com/r/rust/comments/10r0qe6/release_engineering_is_exhausting_so_heres/) too :)

**Comment by MarcoIeni** [comment]
In https://github.com/MarcoIeni/release-plz/pull/538 I explained why I think we should not release binaries. If there are other features from `cargo-dist` that you want in release-plz, please let me know in a separate issue :)

**Comment by MarcoIeni** [comment]
I read the full goreleaser docs.
I have collected some nice ideas in our issues, especially https://github.com/MarcoIeni/release-plz/issues/417.
Most of the features, such as building binaries, or announcing the new release (on twitter, reddit, etc.) are out of scope for the project, because it's easy to achieve them by using other open source projects in combination with release-plz.
For example, I documented how to build binaries [here](https://github.com/MarcoIeni/release-plz/blob/46d0d24cabf4c6ae21828ac428fbf5d549a60372/docs/src/extra/releasing-binaries.md?plain=1#L1).
